sql/core/src/test/scala/org/apache/spark/sql/DatasetBenchmark.scala

```diff
@@ -72,6 +72,47 @@ object DatasetBenchmark {
     benchmark
   }
 
+  def backToBackMapPrimitive(spark: SparkSession, numRows: Long, numChains: Int): Benchmark = {
```

Comment:
This new benchmark looks reasonable, but it's out of the scope of the JIRA. Can you open a new PR to add it? Other changes in this PR LGTM.